xknx/core/config.py

```diff
@@ -50,11 +50,11 @@ def parse_connection(self, doc):
                 and hasattr(doc["connection"], '__iter__'):
             for conn, prefs in doc["connection"].items():
                 try:
-                    if conn.startswith("auto"):
+                    if conn == "auto":
                         self._parse_connection_prefs(ConnectionType.AUTOMATIC, prefs)
-                    elif conn.startswith("tunneling"):
+                    elif conn == "tunneling":
                         self._parse_connection_prefs(ConnectionType.TUNNELING, prefs)
```

Comment:
Shall we raise if `tunneling:` is used without `gateway_ip:`? Or maybe just `logger.warning` and fall back to ConnectionType.AUTOMATIC ?

Reply:
I think we should raise ...